### PR TITLE
Document always setting the resource version in Terraform

### PR DESCRIPTION
**Pull request by hugoShaka**
Prepares the documentation for https://github.com/gravitational/teleport/issues/36773 by always setting the resource version.

We will enforce resource version to be set in v15.

Note for backport: in v13, all `teleport_role` verison must be changed from v7 to v6. The TF reference must be re-generated for v14 and v13.

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Review by marcoandredinis — APPROVED** [review]

File: api/proto/teleport/header/v1/resourceheader.proto [hunk 1]
```diff
@@ -26,7 +26,10 @@ message ResourceHeader {
   string kind = 1;
   // sub_kind is an optional resource sub kind, used in some resources.
   string sub_kind = 2;
-  // version is version.
+  // Version is the API version used to create the resource. It must be
+  // specified. Based on this version, Teleport will apply different defaults on
+  // resource creation or deletion. It ust be an integer prefixed by "v".
```
Comment (marcoandredinis) [inline api/proto/teleport/header/v1/resourceheader.proto:31]:
```suggestion
  // resource creation or deletion. It must be an integer prefixed by "v".
```

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -168,7 +168,10 @@ message ResourceHeader {
   string Kind = 1 [(gogoproto.jsontag) = "kind,omitempty"];
   // SubKind is an optional resource sub kind, used in some resources
   string SubKind = 2 [(gogoproto.jsontag) = "sub_kind,omitempty"];
-  // Version is version
+  // Version is the API version used to create the resource. It must be
+  // specified. Based on this version, Teleport will apply different defaults on
+  // resource creation or deletion. It ust be an integer prefixed by "v".
```
Comment (marcoandredinis) [inline api/proto/teleport/legacy/types/types.proto:173]:
```suggestion
  // resource creation or deletion. It must be an integer prefixed by "v".
```

**Comment by ptgott** [comment]
@hugoShaka Double-checking if we're opening a v15 backport for this one (or already did in another PR). Thanks!

**Comment by hugoShaka** [comment]
This must be backported, but I have found more issues when writing tests for: https://github.com/gravitational/teleport/issues/36773

We might go from "you must always set the version" to "you must set and never change the version".

**Pull request merged**